File: src/Common/Shared.Domain/Events/AuctionFinished.cs
﻿namespace Shared.Domain.Events;

public record AuctionFinished
{
    public Guid AuctionId { get; init; }
    public bool ItemSold { get; init; }

    public string Winner { get; init; }

    public string Seller { get; init; }

    public int? Amount { get; init; }
};

File: frontend/web-app/server/schemas/auctionFinished.ts
import { z } from "zod";

export const auctionFinishedSchema = z.object({
  auctionId: z.string().uuid(),
  itemSold: z.boolean(),
  winner: z.string().nullish(),
  seller: z.string(),
  amount: z.number().nullish(),
});

export type AuctionFinished = z.infer<typeof auctionFinishedSchema>;
